Introduces label component

chuck-sys opened the pull request:
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Fixes #3226 

### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)

Label component is created and is used to replace the old, clunky method of developing labels.

### Review notes:
<!-- (Specify if something in particular should be looked at, or ignored, during review) -->

While looking through the code-base, there were lots of instances where the `label` couldn't be directly applied.

### Steps to test:
- Open Status
- Check that text is displaying properly

<!-- (PRs will only be accepted if squashed into single commit.) -->

status: ready <!-- Can be ready or wip -->

ghost commented:
Hey @cheukyin699, and thank you so much for making your first pull request in status-react! :heart: Please help us make your experience better by filling out this brief questionnaire https://goo.gl/forms/uWqNcVpVz7OIopXg2

<blockquote><img src="https://lh5.googleusercontent.com/wpBxRKA0YohehKsTaptQsaYJNVcNA9m2IOLvxXpSMcamdM8e4q7gyYBQ_yRVvNH-Ez8=w1200-h630-p" width="48" align="right"><div><img src="//ssl.gstatic.com/docs/forms/device_home/android_192.png" height="14"> Google Docs</div><div><strong><a href="https://docs.google.com/forms/d/e/1FAIpQLSewRMgMD6FLe9yO7oCipVn7YTn1HboJc4ZDJ_Va_BsDAemQwQ/viewform?usp=send_form&usp=embed_facebook">Status 1st commit dev survey</a></strong></div><div>First of all thank you so much for taking the time to make your first foray into the world of Status. We want to make sure that your experience with building the future of the decentralized web with us is not only enjoyable but gives you real purpose.

 Please check out our wiki at https://wiki.status.im and our Dev forum at https://discuss.status.im for more in depth longform discussion -- You can also join our Riot at https://chat.status.im which has all of our developers to answer any pressing questions you may have.</div></blockquote>

flexsurfer commented on src/status_im/ui/components/label.cljs:
```diff
@@ -0,0 +1,9 @@
+            [status-im.utils.core :as utils]))
+
+(defn label
+  ([s] (label {} s))
```
why `s` ? maybe something like `i18n-key` ?

flexsurfer commented on src/status_im/ui/components/label.cljs:
```diff
@@ -0,0 +1,9 @@
+
+(defn label
+  ([s] (label {} s))
+  ([attributes s] (components/text {:style attributes} (i18n/label s))))
```
better to use `style` instead `attributes`

flexsurfer reviewed (changes requested):
thanks for the contribution @cheukyin699 , could you please also add default style and merge with the style from arguments, i think most labels have some common styles we can remove from their styles objects and move to default one

chuck-sys commented:
I made a small list of all the styles used in the changes. The most common style is `:font-size 14`, and the next common (it seems) is `:letter-spacing -0.2`. I think that by adding a default style (say, `:font-size 14` for instance), it wouldn't be as clear to the user, which is why I chose not to.

[changes.txt](https://gist.github.com/cheukyin699/d3f496efc7a35308a931c2861c1741f0)

flexsurfer commented:
hey @cheukyin699 great, can we create default style with `:letter-spacing -0.2 :font-size 14` and remove these from styles from your list, thank you!

flexsurfer commented on src/status_im/ui/components/label.cljs:
```diff
@@ -0,0 +1,14 @@
+
+(defn label
+  ([i18n-key] (label {} i18n-key))
+  ([styles i18n-key] (components/text {:style (merge default-styles styles)}
```
i think it should return vector `[`components/text {:style (merge default-styles styles)}

flexsurfer commented:
great job @cheukyin699 , one small comment

goranjovic commented on src/status_im/chat/screen.cljs:
```diff
@@ -20,6 +20,7 @@
             [status-im.chat.views.input.input :as input]
             [status-im.chat.views.toolbar-content :as toolbar-content]
             [status-im.ui.components.animation :as animation]
+            [status-im.ui.components.label :as l]
```
On one hand, we shouldn't break the convention to use the last segment of the namespace as the alias. On the other hand `label/label` looks clumsy. 
How about we put this in `react` namespace and call it `react/label`. I mean, it's that kind of a low level abstraction.
Any thoughts @flexsurfer @jeluard ?

flexsurfer replied:
i'm not sure, because in `react` it means, that it's react native library component, but it's not, hm, btw it's not only this place there it looks clumsy, so we need to find common solution for this case

goranjovic replied:
It wouldn't be the only "slightly higher lever component" than just included from react or a library. e.g. there is `icon` as a slight abstraction over `image`.

goranjovic commented on src/status_im/ui/screens/wallet/views.cljs:
```diff
@@ -36,17 +37,15 @@
                   :accessibility-label :total-amount-currency-text}
       (:code currency)]]
-    [react/text {:style styles/total-value} (i18n/label :t/wallet-total-value)]]])
+    [l/label styles/total-value :wallet-total-value]]])
```
I noticed all i18n keywords are de-qualified. Why is that?

flexsurfer replied:
it was my initial idea, what's the reason to qualify them? it's difficult to search translation

goranjovic replied:
fair enough

jeluard commented:
Hi @cheukyin699 ! Thanks for your contribution!

After discussing with @flexsurfer and @goranjovic we decided it would be better to have the following syntax for the component:

```cljs
[react/i18n-text {:style style :key :wallet-total-value}]
```

This component would be defined in the `react` namespace. 

Sorry for the inconvenience!

chuck-sys commented:
@jeluard Should default styles be added?

jeluard commented:
@cheukyin699 It's fine to have it part of the `i18n-text` component for now. Thanks!

chuck-sys committed:
Introduces label component

Label component located inside `status-im.ui.components.react`
namespace, called i18n-text. It replaces the need for the standard way
of creating labels using both `react/text` and `i18n/label`.

Signed-off-by: Goran Jovic <goranjovic@gmail.com>

jeluard commented on src/status_im/ui/screens/wallet/views.cljs:
```diff
@@ -36,17 +36,18 @@
-      [react/text {:style styles/backup-seed-phrase-description}
-       (i18n/label :t/wallet-backup-seed-description)]]
+      [react/i18n-text {:style styles/backup-seed-phrase-title
+                        :key   :t/wallet-backup-seed-title}]
```
Are those `:t/` still expected?

chuck-sys replied:
Good point.

Also, though it should be a separate issue, we can remove `:t/` for any instance of it in `(i18n/label :t/wallet-manage-assets)`, since that part is removed with `name`.

flexsurfer commented on src/status_im/ui/screens/browser/views.cljs:
```diff
@@ -113,7 +111,7 @@
          :injected-java-script                  js-res/webview-js}]
        [react/view styles/background
-        [react/text (i18n/label :t/enter-dapp-url)]])
+        [react/i18n-text {:style {} :key :enter-dapp-url}]])
```
it will work fine without `:style {} `

flexsurfer commented on src/status_im/ui/screens/network_settings/edit_network/views.cljs:
```diff
@@ -47,7 +47,7 @@
           :default-value   (get-in manage-network [:url :value])
           :on-change-text  #(re-frame/dispatch [:network-set-input :url %])}]
-        [react/text (i18n/label :t/network-chain)]
+        [react/i18n-text {:style {} :key :network-chain}]
```
same

lukaszfryc commented:
@cheukyin699 

1) I noticed one missing translation error:

<img src="https://user-images.githubusercontent.com/7532782/42627112-5c755e20-85cc-11e8-8436-67c8c13ff76a.png" width="200">

Could you make sure that this is the only label that had this issue?


2) Broken padding between wallet phrase and "Sign the transaction by entering your password..." text. See the screenshot below. This PR build is on the left, on the right is version from develop. Could you check why the padding is so small in this PR?

<img src="https://user-images.githubusercontent.com/7532782/42627743-331dd56e-85ce-11e8-8b8f-096c963c6b30.png" width="400">

chuck-sys commented:
@lukaszfryc

1: I'm pretty sure that is the only issue - a typo. I took a look at the diff, and could not find any more typos.

2: Another typo that caused a style to not load. Had them all checked and could not find more typos.

lukaszfryc commented:
@cheukyin699 looks great now. Thanks!

jeluard commented:
@cheukyin699 Sorry looks like your PR now has some conflicts.. Would you mind resolving it so that I can merge it?
